pkg/submitter/submitter.go

```diff
@@ -0,0 +1,504 @@
+						s.submitCount.Inc()
+						return
+					}
+				}
+			}
+		}
+	}(ctx, result)
+}
+
+func (s *Submitter) Submit(ctx context.Context, result *mining.Result) (*types.Transaction, error) {
+	challenge := result.Work.Challenge
+	nonce := result.Nonce
+	s.currentChallenge = challenge
+	s.currentNonce = nonce
+
+	// The submit contains values for 5 data IDs so add them here.
+	for i := 0; i < 5; i++ {
+		valKey := fmt.Sprintf("%s%d", db.QueriedValuePrefix, challenge.RequestIDs[i].Uint64())
+		m, err := s.proxy.BatchGet([]string{valKey})
+		if err != nil {
+			return nil, errors.Wrapf(err, "retrieve pricing data for current request id")
+		}
+		val := m[valKey]
+		var value *big.Int
+		if len(val) == 0 {
+			jsonFile, err := os.Open(s.cfg.ManualDataFile)
+			if err != nil {
+				return nil, errors.Wrapf(err, "manualData read Error")
+			}
+			defer jsonFile.Close()
+			byteValue, _ := ioutil.ReadAll(jsonFile)
+			var result map[string]map[string]uint
+			_ = json.Unmarshal([]byte(byteValue), &result)
+			_id := strconv.FormatUint(challenge.RequestIDs[i].Uint64(), 10)
+			val := result[_id]["VALUE"]
+			if val == 0 {
+				return nil, errors.Errorf("retrieve pricing data for current request id")
+			}
+			value = big.NewInt(int64(val))
+		} else {
+			value, err = hexutil.DecodeBig(string(val))
+			if err != nil {
+				if challenge.RequestIDs[i].Uint64() > tracker.MaxPSRID() {
+					level.Error(s.logger).Log(
+						"msg", "decoding price value prior to submitt ing solution",
+						"err", err,
+					)
+					if len(val) == 0 {
+						level.Error(s.logger).Log("msg", "0 value being submitted")
+						s.currentValues[i] = big.NewInt(0)
+					}
+					continue
+				}
+				return nil, errors.Errorf("no value in database,  reg id:%v", challenge.RequestIDs[i].Uint64())
+			}
+		}
+		s.currentValues[i] = value
+	}
+
+	// Submit the solution.
+	tx, err := s.submitter.Submit(ctx, s.proxy, "submitSolution", s.submit)
+	if err == nil {
+		return tx, nil
+	}
+	level.Error(s.logger).Log("msg", "submit solution", "pubkey", s.submitter.Address().String())
+	return nil, errors.New("submitting solution txn by any account")
+}
+
+func (s *Submitter) getMinerStatus() int64 {
+	// Check if the staked account is in dispute before sending a transaction.
+	v, dispute := s.IsInDispute()
+	if dispute {
+		return -1
+	}
+	if len(v) == 0 {
+		level.Error(s.logger).Log("msg", "no status result. this usually means no connection to the DB")
+		return -1
+	}
+
+	status, _ := hexutil.DecodeBig(string(v))
+	return status.Int64()
+}
+
+func (s *Submitter) submit(ctx context.Context, contract tellorCommon.ContractInterface) (*types.Transaction, error) {
+
+	txn, err := contract.SubmitSolution(
+		s.currentNonce,
+		s.currentChallenge.RequestIDs,
+		s.currentValues)
+	if err != nil {
+		return nil, err
+	}
+
+	return txn, err
+}
+
+func (s *Submitter) lastSubmit() (time.Duration, *time.Time, error) {
+	address := "000000000000000000000000" + s.account.Address.Hex()
```

Comment:
```suggestion
	address := "000000000000000000000000" + s.account.Address.Hex()[2:]
```
don't need the `0x` from the address.